decoder_test.go

```diff
@@ -362,18 +340,11 @@ func TestReadString(t *testing.T) {
 		}
 
 		str, err := d.readString()
-		if !reflect.DeepEqual(err, tt.err) {
-			t.Errorf("invalid error '%v' '%v'", err, tt.err)
-		}
-
-		if str != tt.string {
-			t.Errorf("invalid string '%v' '%v'", str, tt.string)
-		}
+		assert.Truef(t, reflect.DeepEqual(err, tt.err), "invalid error '%v' '%v'", err, tt.err)
+		assert.Equal(t, tt.string, str)
 
 		hash := d.h.Sum(nil)
-		if !reflect.DeepEqual(hash, tt.hash[:]) {
-			t.Errorf("invalid hash '%v' '%v'", hash, tt.hash)
-		}
+		assert.Truef(t, reflect.DeepEqual(hash, tt.hash[:]), "invalid hash '%v' '%v'", hash, tt.hash)
```

Comment:
Please consider using assert.Equal and put expected and actual arguments in a correct order.